packages/core/admin/admin/src/content-manager/components/BlocksInput/EditorLayout.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+
+import { Box, Flex, FocusTrap, Portal, IconButton, InputWrapper } from '@strapi/design-system';
+import { useLockScroll, pxToRem } from '@strapi/helper-plugin';
+import { Collapse } from '@strapi/icons';
+import { useIntl } from 'react-intl';
+import styled from 'styled-components';
+
+import { getTranslation } from '../../utils/translations';
+
+const CollapseIconButton = styled(IconButton)`
+  position: absolute;
+  bottom: ${pxToRem(12)};
+  right: ${pxToRem(12)};
+`;
+
+interface EditorLayoutProps {
+  children: React.ReactNode;
+  isExpandMode: boolean;
+  error?: string;
+  onCollapse: () => void;
+  disabled: boolean;
+  ariaDescriptionId: string;
+}
+
+const EditorLayout = ({
+  children,
+  isExpandMode,
+  error,
+  disabled,
+  onCollapse,
+  ariaDescriptionId,
+}: EditorLayoutProps) => {
+  const { formatMessage } = useIntl();
+
+  useLockScroll({ lockScroll: isExpandMode });
+
+  if (isExpandMode) {
+    return (
+      <Portal role="dialog" aria-modal={false}>
+        <FocusTrap onEscape={onCollapse} skipAutoFocus>
```

Comment:
I assume you checked that it wasn't possible to use any of the modal components from the DS here?

Reply:
I'm guessing ModalLayout?